### ISSUE TITLE
Creating OpenSearch Dashboard Plugin

### ISSUE THREAD

**Issue by Rohr2827** [opensearch-project/OpenSearch-Dashboards#5755]
I am creating a personal plugin for OpenSearch and have followed the DEVELOPER_GUIDE.md but am uncertain of how or where to gather more information on easy and complex tasks to improve my plugin. Currently I am creating the base plugin with the code 

**node scripts/generate_plugin --name my_plugin_name –y**  Create a default plugin
**yarn plugin-helpers build**                                                       Builds plugin

After this I have developed a relatively complex reporting plugin using the API with front and backend code that uses JS and typescript but am uncertain if I am doing everything the easiest way or even the desired way. In addition, I am not certain where to even start with potentially more complex tasks like getting the current logged in user info, utilizing the built in emailing, or how to change the breadcrumbs that act as tabs in your header. 

![Capture](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/140717036/0a979154-38f3-44ba-abc4-005961a1ce1e)

In summary, is there any specific step by step guide that encompasses building a plugin from the ground up like the Reporting, Alerting, Maps, Notifications, Index Management, Snapshot Management, or Security plugins or is it something I must discover on my own?

**Comment by abbyhu2000** [comment]
@manasvinibs share the links related to the topics.

**Comment by manasvinibs** [comment]
Here are some of the useful links to understand plugin development lifecycle in OpenSearch Dashboards:
How to develop simple plugin - https://cptnb.github.io/opensearch-dashboards-dev-docs/plugins/demo-plugin-tutorial/ 
Understanding plugin life cycle - https://opensearch.org/blog/dashboards-plugins-intro/
How to install plugin - https://cptnb.github.io/opensearch-dashboards-dev-docs/plugins/install/
